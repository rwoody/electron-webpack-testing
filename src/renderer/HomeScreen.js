import React, {Component} from 'react'

export class HomeScreen extends Component {
  render() {
    return (
      <div>
        <p>Hello from HomeScreen</p>
      </div>
    )
  }
}