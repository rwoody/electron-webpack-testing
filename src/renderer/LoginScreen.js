import React, {Component} from 'react'

export class LoginScreen extends Component {
  render() {
    return (
      <div>
        <p>Hello from LoginScreen</p>
      </div>
    )
  }
}