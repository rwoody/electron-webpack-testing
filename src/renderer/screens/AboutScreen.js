import React, {Component} from 'react'

export class AboutScreen extends Component {
  render() {
    return (
      <div>
        <p>Hello from AboutScreen</p>
      </div>
    )
  }
}

export default AboutScreen