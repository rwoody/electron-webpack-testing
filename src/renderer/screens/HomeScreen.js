import PropTypes from 'prop-types'
import React, {Component} from 'react'
import {graphql} from 'react-apollo'
import gql from 'graphql-tag'
import {compose} from 'recompose'

import {renderWhileLoading} from '../hoc/renderWhileLoading'
import {Loading} from '../ui/loading'

export const HOME_QUERY = gql`
  {
    home {
      path
      title
      items
      counter
    }
  }
`

class HomeScreen extends Component {
  static propTypes = {
    data: PropTypes.object
  }

  render() {
    const {home} = this.props
    return (
      <div>
        <h1>{home.title}</h1>
        {home.items && home.items.length > 0 &&
          <ul>
            {home.items.map(i => <li key={i}>{i}</li>)}
          </ul>
        }
        {home.items && home.items.length === 0 &&
          <p>No items found</p>
        }
        <div style={{display: 'flex', flexDirection: 'column'}}>
          <p>Counter: {home.counter}</p>
          <div style={{flexDirection: 'row'}}>
            <button>Inc</button>
            <button>Dec</button>
          </div>
        </div>
        <pre>Path: {home.path}</pre>
      </div>
    )
  }
}

HomeScreen = renderWhileLoading(Loading)(HomeScreen)

export {HomeScreen}
export const withHome = graphql(HOME_QUERY, {
  props: ({data}) => ({...data})
})
export default withHome(HomeScreen)
