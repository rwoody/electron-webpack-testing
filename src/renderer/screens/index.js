export {default as Home} from './HomeScreen'
export {default as About} from './AboutScreen'
export {default as Login} from './LoginScreen'