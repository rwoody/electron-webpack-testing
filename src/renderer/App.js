import React, {Component} from 'react'
import {
  Link,
  Route,
  Switch
} from 'react-router-dom'

import * as Screens from './screens'

export class App extends Component {
  render() {
    return (
      <div>
        <ul>
          <li><Link to='/'>Home</Link></li>
          <li><Link to='/about'>About</Link></li>
          <li><Link to='/login'>Login</Link></li>
        </ul>
        <Switch>
          <Route exact path='/' component={Screens.Home} />
          <Route path='/about' component={Screens.About} />
          <Route path='/login' component={Screens.Login} />
        </Switch>
      </div>
    )
  }
}