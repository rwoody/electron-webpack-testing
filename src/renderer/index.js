import React, {Component} from 'react'
import ReactDOM from 'react-dom'
import {HashRouter as Router} from 'react-router-dom'
import {ApolloClient} from 'apollo-client'
import {ApolloProvider} from 'react-apollo'
import {HttpLink} from 'apollo-link-http'
import {InMemoryCache} from 'apollo-cache-inmemory'
import {App} from './App'

const client = new ApolloClient({
  link: new HttpLink({uri: 'https://0v9rj7m375.lp.gql.zone/graphql'}),
  cache: new InMemoryCache(),
})

const components = (
  <ApolloProvider client={client}>
    <Router>
      <App />
    </Router>
  </ApolloProvider>
)

ReactDOM.render(
  components,
  document.getElementById('app')
)
