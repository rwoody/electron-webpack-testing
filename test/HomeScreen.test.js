import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import { print } from 'graphql/language/printer'
import {MockedProvider} from 'react-apollo/test-utils'
// import { addTypenameToDocument } from 'apollo-utilities'
Enzyme.configure({ adapter: new Adapter() })

import {HOME_QUERY, withHome, HomeScreen} from '../src/renderer/screens/HomeScreen'
import {
  home_full,
  home_no_items,
} from './__mocks__/data'

// const query = addTypenameToDocument(HOME_QUERY)
const query = HOME_QUERY

describe('HomeScreen', () => {

  test('has a known query shape', () => {
    expect(print(HOME_QUERY)).toMatchSnapshot()
  })

  test('has a loading state', done => {
    class Container extends React.Component {
      componentWillMount() {
        expect(this.props.loading).toBe(true)
      }
      componentWillReceiveProps(next) {
        expect(next.loading).toBe(false)
        done()
      }
      render() {
        return null
      }
    }

    const ContainerWithData = withHome(Container)
    const mocks = [
      {
        request: { query },
        result: { data: { home: home_full } },
      },
    ]
    renderer.create(
      <MockedProvider mocks={mocks}>
        <ContainerWithData />
      </MockedProvider>,
    )
  })

  describe('Component', () => {
    test('handles loading state', () => {
      const output = renderer.create(<HomeScreen loading />);
      expect(output.toJSON()).toMatchSnapshot();
    })
    test('returns markup', () => {
      const output = renderer.create(<HomeScreen home={home_full} />);
      expect(output.toJSON()).toMatchSnapshot();
    })
    test('returns markup when no items', () => {
      const output = renderer.create(<HomeScreen home={home_no_items} />);
      expect(output.toJSON()).toMatchSnapshot();
    })
  })

})
