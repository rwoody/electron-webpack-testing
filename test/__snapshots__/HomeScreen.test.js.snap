// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Component handles loading state 1`] = `
<p>
  Loading...
</p>
`;

exports[`HomeScreen Component returns markup 1`] = `
<div>
  <h1>
    Electron App - Test page
  </h1>
  <ul>
    <li>
      Test Item 1
    </li>
    <li>
      Test Item 2
    </li>
  </ul>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
      }
    }
  >
    <p>
      Counter: 
      3
    </p>
    <div
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <button>
        Inc
      </button>
      <button>
        Dec
      </button>
    </div>
  </div>
  <pre>
    Path: 
    /
  </pre>
</div>
`;

exports[`HomeScreen Component returns markup when no items 1`] = `
<div>
  <h1>
    Electron App - Test page
  </h1>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
      }
    }
  >
    <p>
      Counter: 
      3
    </p>
    <div
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <button>
        Inc
      </button>
      <button>
        Dec
      </button>
    </div>
  </div>
  <pre>
    Path: 
    /
  </pre>
</div>
`;

exports[`HomeScreen has a known query shape 1`] = `
"{
  home {
    path
    title
    items
    counter
  }
}
"
`;
